### PR TITLE
Prevent penalty pegs from counting against peg limit

### PR DESCRIPTION
**Pull request by jimcullenaus**
This PR does **not** apply any penalty to the results/winners for penalty pegs. That will have to be done in a future PR. For this reason, I am not closing the Issue this PR relates to.

Related to #91

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "PockyBot",
-	"version": "1.5.1",
+	"version": "1.6.2",
```
Comment (Evangelinexx) [inline package.json]: Technically if we are following semver, we should bump once per **release**, but I'm fine with this anyway.

File: src/lib/database/pocky-db.ts [hunk 2]
```diff
@@ -54,7 +55,11 @@ export default class PockyDB implements PockyDbInterface {
 			return dbConstants.pegError;
 		}
 
-		if (!senderHasPegs) {
+		// TODO: make this faster and less repetitive by refactoring getting the list of penaltyKeywords.
+		const penaltyKeywords = this.config.getStringConfig('penaltyKeyword');
+
+		if (!senderHasPegs
+			&&!penaltyKeywords.some(keyword => comment.toLowerCase().includes(keyword.toLowerCase()))) {
```
Comment (Evangelinexx) [inline src/lib/database/pocky-db.ts]: Does this work?

File: package.json (hunk 1 above)
Comment (Evangelinexx) [inline package.json]: Technically if we are following semver, we should bump once per release, but I'm fine with this anyway.

File: src/lib/database/pocky-db.ts (hunk 2 above)
Comment (Evangelinexx) [inline src/lib/database/pocky-db.ts]: Should move this sort of logic to hasSparePegs. Also it doesn't work.

Comment (jimcullenaus) [reply src/lib/database/pocky-db.ts]: Yeah will do.

File: package.json (hunk 1 above)
Comment (jimcullenaus) [reply package.json]: Yeah good point. But uhh, are we able to deploy during the development process without bumping the patch version?

File: src/lib/database/pocky-db.ts (hunk 2 above)
Comment (Lauraducky) [inline src/lib/database/pocky-db.ts]: Shouldn't this also check that there are no 'good' keywords in the message, since those don't count as shame pegs?

Comment (jimcullenaus) [reply src/lib/database/pocky-db.ts]: Yes it should. Forgot to update this bit when switching from a default-penalty to default-good.

File: package.json (hunk 1 above)
Comment (Lauraducky) [reply package.json]: yes that is possible @jimcullenaus

File: src/lib/database/pocky-db.ts [hunk 3]
```diff
@@ -41,6 +41,7 @@ export default class PockyDB implements PockyDbInterface {
 	 *         2 on error
 	 */
 	async givePegWithComment(comment : string, receiver : string, sender = 'default_user') : Promise<number> {
```
Comment (Evangelinexx) [inline src/lib/database/pocky-db.ts]: This should let you give shame pegs even if all your pegs are spent.

ie. 
```
if (!senderHasPegs || !isShamePeg) {
 //block peg
}
//send peg
```

Comment (Evangelinexx) [reply src/lib/database/pocky-db.ts]: You could refactor senderHasPegs to senderCanSendPeg instead, to cover this case.

**Pull request merged**